components/director/internal/domain/label/repository_test.go

```diff
@@ -511,7 +511,115 @@ func TestRepository_List(t *testing.T) {
 		objID := "foo"
 
 		// WHEN
-		_, err := repo.List(context.TODO(), "tenant", objType, objID)
+		_, err := repo.ListForObject(context.TODO(), "tenant", objType, objID)
+		// THEN
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "unable to fetch database from context")
+	})
+}
+
+func TestRepository_ListByKey(t *testing.T) {
+	t.Run("Success - Label for Application and Runtime", func(t *testing.T) {
+		// GIVEN
+		tnt := "tenant"
+		labelKey := "foo"
+		objType := model.RuntimeLabelableObject
+		rtmObjID := "foo"
+		appObjID := "bar"
+		inputItems := []label.Entity{
+			{ID: "1", TenantID: tnt, Key: labelKey, Value: "test1", RuntimeID: sql.NullString{Valid: true, String: rtmObjID}},
+			{ID: "2", TenantID: tnt, Key: labelKey, Value: "test2", AppID: sql.NullString{Valid: true, String: appObjID}},
+		}
+		expected := []*model.Label{
+			{ID: "1", Tenant: tnt, Key: labelKey, Value: "test1", ObjectType: objType, ObjectID: rtmObjID},
+			{ID: "2", Tenant: tnt, Key: labelKey, Value: "test2", ObjectType: model.ApplicationLabelableObject, ObjectID: appObjID},
+		}
+
+		mockConverter := &automock.Converter{}
+		defer mockConverter.AssertExpectations(t)
+		mockConverter.On("FromEntity", inputItems[0]).Return(*expected[0], nil).Once()
+		mockConverter.On("FromEntity", inputItems[1]).Return(*expected[1], nil).Once()
+
+		repo := label.NewRepository(mockConverter)
+
+		db, dbMock := mockDatabase(t)
+		defer func() {
+			require.NoError(t, dbMock.ExpectationsWereMet())
+		}()
+
+		escapedQuery := regexp.QuoteMeta(`SELECT "id", "tenant_id", "key", "value", "app_id", "runtime_id" 
```

Comment:
Still, the formatting looks a bit weird.

Reply:
done